include/SDL_syswm.h

```diff
@@ -289,8 +289,9 @@ struct SDL_SysWMinfo
         {
             struct wl_display *display;             /**< Wayland display */
             struct wl_surface *surface;             /**< Wayland surface */
-            struct wl_shell_surface *shell_surface; /**< Wayland shell_surface (window manager handle) */
             struct wl_egl_window *egl_window;       /**< Wayland EGL window (native window) */
+            void *shell_surface;                    /**< DEPRECATED Wayland shell_surface (window manager handle) */
```

Comment:
For ABI compatibility this has to stay above egl_window, sadly.

Reply:
Ah yes, too much haste sorry.